homeassistant/components/mqtt/device_tracker.py

```diff
@@ -151,16 +156,44 @@ def _process_update_extra_state_attributes(
         self, extra_state_attributes: dict[str, Any]
     ) -> None:
         """Extract the location from the extra state attributes."""
-        self._attr_latitude = extra_state_attributes.get(ATTR_LATITUDE)
-        self._attr_longitude = extra_state_attributes.get(ATTR_LONGITUDE)
         if (
             ATTR_LATITUDE in extra_state_attributes
             or ATTR_LONGITUDE in extra_state_attributes
         ):
-            # Reset manual set location
+            latitude: float | None
+            longitude: float | None
+            gps_accuracy: int
```

Comment:
We allow float below, I think this should be typed as float

Reply:
MyPy does not like it:

```
homeassistant/components/mqtt/device_tracker.py:188: error: Incompatible types in assignment (expression has type "int | float", variable has type "int")  [assignment]
```

Shall I override? The architecture discussion is still going on.